core/src/main/scala/zio/config/ConfigDescriptor.scala

```diff
@@ -170,15 +171,66 @@ object ConfigDescriptor {
   val float: ConfigDescriptor[String, String, Float] =
     ConfigDescriptor.Source(ConfigSource.empty, PropertyType.FloatType) ?? "value of type float"
 
+  def first[K, V, A](path: K)(desc: ConfigDescriptor[K, V, A]) =
+    list(path)(desc)
+      .xmapEither[A](_.headOption.fold[Either[String, A]](Left("Element is missing"))(Right(_)), v => Right(v :: Nil))
+
   def float(path: String): ConfigDescriptor[String, String, Float] = nested(path)(float)
 
   val int: ConfigDescriptor[String, String, Int] =
     ConfigDescriptor.Source(ConfigSource.empty, PropertyType.IntType) ?? "value of type int"
 
   def int(path: String): ConfigDescriptor[String, String, Int] = nested(path)(int)
 
-  def list[K, V, A](desc: ConfigDescriptor[K, V, A]): ConfigDescriptor[K, V, List[A]] =
-    ConfigDescriptor.Sequence(desc)
+  /**
+   * Leaks out path in simple cases, i.e.
+   * `list(string(path))` == `list(path)(string)` == `nested(path)(listOrSingle(string))`
```

Comment:
I'm not sure these equalities should hold.

This config descriptor:

```
[foo = 2, foo = 3]
```

is different than this one:

```
foo = [2, 3]
```

and we should preserve this distinction.

Reply:
`list` keeps backward compatibility with all the docks. See `listStrict` below for.

Reply:
I think we should not keep backward compatibility, because the old way was not precise.

Reply:
done